Gridpane styling

Markvanderuit opened the pull request:
Timeline now has 'subpanes', which basically means the grid lines don't show up on every grid. I also added two new scrollable grids of the exact same height, number of grids, etc as the normal timeline. These scroll together with the main timeline, so they're all basically one whole. The left one will be used for showing a more detailed counter, and the right one will be used for the director timeline.

Markvanderuit committed:
Fixed issue where gridpane could not resize properly due to incorrect use of variables.

Markvanderuit committed:
FIxed issue where bottom dragging teleported.

Markvanderuit committed:
Enabled subgrids for grid.

Markvanderuit committed:
Moved all styling to stylesheet.css.

Markvanderuit committed:
Started work on vertical counter bar. Will finish when I get home.

Markvanderuit committed:
Renamed panes of the main timeline to better distinguish it from other panes.
Added classes and panes for a director timeline.
Added classes and panes for a counter timeline.

Markvanderuit committed:
Checkstyle fixes.

Markvanderuit committed:
The three srollpanes now bidirectionally sync scrolling settings.

Markvanderuit committed:
Checkstyle fixes.

Markvanderuit committed:
Merge branch 'develop' into gridpane-styling

geenen124 commented on src/main/java/gui/CounterGridPane.java:
```diff
@@ -0,0 +1,52 @@
+
+/**
+ * Class that represents the grid pane in the counter bar or timeline.
+ * Created by markv on 5/4/2016.
```
Replace this with an author annotation

geenen124 commented on src/main/java/gui/DirectorGridPane.java:
```diff
@@ -0,0 +1,53 @@
+
+/**
+ * Class that represents the grid pane in the director timeline.
+ * Created by markv on 5/4/2016.
```
Same here

geenen124 commented on src/main/java/gui/DirectorGridPane.java:
```diff
@@ -0,0 +1,53 @@
+            rc.setVgrow(Priority.ALWAYS);
+            this.getRowConstraints().add(rc);
+        }
+
```
Extra space?

geenen124 commented on src/main/java/gui/RootLeftArea.java:
```diff
@@ -1,12 +1,25 @@
+
 /**
  * Created by markv on 4/26/2016.
+ * Mwahahaha I created this class, losers.
```
?

geenen124 commented on src/main/java/gui/TimelinesGridPane.java:
```diff
@@ -94,13 +93,22 @@ public void removeCameraShotBlock(CameraShotBlock block) {
+                    c = 2;
+                } else {
+                    pane.setStyle("-fx-border-color: gray;"
+                            + "-fx-border-width: 0px 1px 0px 1px;");
```
Consider deferring to the stylesheet here?

Markvanderuit committed:
Implemented Alex' recommended changes.

bart-de-jonge commented on src/main/java/gui/CounterGridPane.java:
```diff
@@ -0,0 +1,52 @@
+    @Getter
+    private int numberOfCounts;
+
+    // Height and width of count and timelines. Note that timelineWidth is a minimum.
```
this commment talks about timelineWidth but there is no such field present

bart-de-jonge commented on src/main/java/gui/CounterGridPane.java:
```diff
@@ -0,0 +1,52 @@
+package gui;
```
Shouldn't this class be named something like CameraGridPane?

Markvanderuit replied:
The class will represent the small sidebar showing counter numbers. Basically this http://puu.sh/oGcW2/c7c957ff6d.jpg
So it has nothing to do with the camera. I'll add comments to better explain its function.

bart-de-jonge replied:
My bad, I had it mistaken for another class. Then I think the name is good

bart-de-jonge commented on src/main/java/gui/RootCenterArea.java:
```diff
@@ -1,62 +1,155 @@
+
+        counterAndDirectorPane = new HBox();
+        setAlignment(counterAndDirectorPane, Pos.CENTER_LEFT);
+        counterAndDirectorPane.setMaxWidth(200);
```
maybe use a constant here to make it easily tweakable?

bart-de-jonge commented on src/main/java/gui/RootCenterArea.java:
```diff
@@ -1,62 +1,155 @@
+        mainTimeLineAnchorPane.setRightAnchor(mainTimeLineGridPane, 0.0);
+        mainTimeLineAnchorPane.setTopAnchor(mainTimeLineGridPane, 0.0);
+        mainTimeLineAnchorPane.getChildren().add(mainTimeLineGridPane);
+        mainTimeLineAnchorPane.setMinHeight(1000);
```
same here, maybe use constants. Same holds for some more numbers in this and the following methods

bart-de-jonge commented on src/main/java/gui/RootCenterArea.java:
```diff
@@ -1,62 +1,155 @@
+        ScrollBar scrollbar = new ScrollBar();
+        scrollbar.setMin(0);
+        scrollbar.setMax(1);
+        scrollbar.maxWidthProperty().bind(widthProperty().subtract(200));
```
This depends on the min width for the directortimeline i suppose, extra reason to use constants

bart-de-jonge commented on src/main/java/gui/RootPane.java:
```diff
@@ -45,7 +45,7 @@ public void start(Stage primaryStage) throws Exception {
-        scene.getStylesheets().add("testcss.css");
+        scene.getStylesheets().add("stylesheet.css");
         primaryStage.setScene(scene);
         primaryStage.setTitle("Hoi ben een titel lol.");
```
Could you change this to a more suitable title and maybe make a constant to make it easy accesible/tweakable

Markvanderuit replied:
Can't right now. CSS is applied to the entire scene for now, it doesn't seem to work properly when a stylesheet is applied to a single object only.

Furthermore, the stylesheet is applied recursively to every single object, and is accessible (using getStyleSheet) from every (every!) single object, already.

bart-de-jonge replied:
These line note apply to the lines they are pinned to. I meant the title "hoi ben een titel lol." And I think it would be nice if that would be a string constant in the top of this class

bart-de-jonge commented on src/main/java/gui/RootPane.java:
```diff
@@ -45,7 +45,7 @@ public void start(Stage primaryStage) throws Exception {
         // created. Also has some useful settings.
         //TODO: change this line of code.
```
What is this todo supposed to do? The description is a bit weird

Markvanderuit replied:
Good question, previous code review someone insisted on it being there. I've removed it.

bart-de-jonge commented on src/main/java/gui/TimelinesGridPane.java:
```diff
@@ -94,13 +93,20 @@ public void removeCameraShotBlock(CameraShotBlock block) {
         for (int i = 0; i < numberOfTimelines; i++) {
+            c = 1;
             for (int j = 0; j < numberOfCounts; j++) {
                 SnappingPane pane = new SnappingPane(j, i, 200, countHeight);
```
constant

bart-de-jonge commented on src/main/java/gui/TimelinesGridPane.java:
```diff
@@ -50,9 +50,8 @@ public TimelinesGridPane(int numberOfTimelines, int numberOfCounts,
-        // side-panes overlap it without messing up the scrolling.
-        //this.setPadding(new Insets(5, 5, 5, 5));
+        // offset to the left
+        this.setPadding(new Insets(0, 0, 0, 200));
```
constant

bart-de-jonge commented on src/main/java/gui/TimelinesGridPane.java:
```diff
@@ -94,13 +93,20 @@ public void removeCameraShotBlock(CameraShotBlock block) {
                 panes.add(pane);
-                pane.setStyle("-fx-border-color: gray;"
-                        + "-fx-border-width: 0.1px 1px 0.1px 1px;");;
+                if (c > 4) {
```
Maybe add a little comment about what style pattern the grid gets from this loop

bartdejonge1996 commented:
I see a lot of duplicate code between CountGridPane and DirectorGridPane. I think you should make an abstract superclass that contains these functionalities and subclass that to prevent all this duplicate code.

Ezlo123 commented:
@bartdejonge1996 I already warned about the duplicate code. I don't think it would be wise to do that now, as they are just quick templates so martijn and I can make the counter and director grids, which will be totally completely different. They should diverge completely in the next few builds.

If I'm gonna give them a superclass, I should probably include the big timeline with this as well. What do you think?

Markvanderuit committed:
Implemented Barts' recommended fixes.

Markvanderuit committed:
Merge branch 'gridpane-styling' of https://github.com/bartdejonge1996/goto-fail into gridpane-styling

# Conflicts:
#	src/main/java/gui/TimelinesGridPane.java

Markvanderuit committed:
Added some extra comments

bartdejonge1996 commented:
You warned about the initialisations and I think that is fine for now, but the classes DirectorGridPane and CounterGridPane are also really alike and I doubt their structure will change a lot (the styling off course will). Therefore I think a superclass would be suitable. And yes the big timeline could be included in that as well

Markvanderuit committed:
Subclassed all three gridpane classes into ScrollableGridPane class.

Tried to do the same for the Scrollpane they're in, but backed out halfway, unnecessary for now.

Ezlo123 commented:
@bartdejonge1996 big skaboom timeline now included in superclass as well. I was going to do the same for the scrollables they're in, but I backed out, it made things too complicated for now.

bart-de-jonge commented on src/main/java/gui/TimelinesGridPane.java:
```diff
@@ -97,10 +60,10 @@ public void removeCameraShotBlock(CameraShotBlock block) {
-            for (int j = 0; j < numberOfCounts; j++) {
-                SnappingPane pane = new SnappingPane(j, i, 200, countHeight);
+            for (int j = 0; j < getNumberOfVerticalGrids(); j++) {
+                SnappingPane pane = new SnappingPane(j, i, 200, getVerticalElementSize());
```
Please use a constant for the 200

Markvanderuit replied:
Turns out, the snappingpane class doesn't even need width and height, so I just removed them completely.

bart-de-jonge commented on src/main/java/gui/RootCenterArea.java:
```diff
@@ -85,7 +85,8 @@ private void initMainTimeLinePane() {
         // main timeline panes
         mainTimelineScrollpane = new ScrollPane();
         mainTimeLineAnchorPane = new AnchorPane();
-        mainTimeLineGridPane = new TimelinesGridPane(numberOfTimelines, numberOfCounts, 1000, 1000);
```
I think you should use constants here and the following 3/4 methods on the 10's, 100's and 1000's as stated in previous comments

Markvanderuit replied:
Removed most of them as they were unnecessary, and changed the rest to constants.

Markvanderuit committed:
Moved all gui classes into respective subpackages.

If you feel there is a better way to divide them, feel free to give a suggestion!

Markvanderuit committed:
Implemented rest of remaining feedback.

bartdejonge1996 commented:
Looks good to me!

Markvanderuit committed:
My bad.

martijn9612 commented:
Nice stuff!

Markvanderuit committed:
Merge branch 'develop' into gridpane-styling

# Conflicts:
#	src/main/java/control/TimelineController.java
